test/e2e/common/pod_resize.go

```diff
@@ -210,20 +215,39 @@ func verifyPodStatusResources(pod *v1.Pod, tcInfo []TestContainerInfo) {
 	for i, c := range pod.Status.ContainerStatuses {
 		csMap[c.Name] = &pod.Status.ContainerStatuses[i]
 	}
+
 	for _, ci := range tcInfo {
 		cs, found := csMap[ci.Name]
 		framework.ExpectEqual(found, true)
-		tc := makeTestContainer(ci)
-		framework.ExpectEqual(cs.Resources, tc.Resources)
+
+		// Fixed the testing error for container specifying memory only, added by chenw.
+		// When creating a container in a pod without specifying the CPUReq,  its Resource.CPUReq automatically became "2m"
+		if ci.Resources.CPUReq == "" {
+			ci.Resources.CPUReq = NotSpecifiedCPUReq
+
+			tc := makeTestContainer(ci)
+			framework.ExpectEqual(cs.Resources, tc.Resources)
+
+			ci.Resources.CPUReq = ""
+		} else {
+			tc := makeTestContainer(ci)
+			framework.ExpectEqual(cs.Resources, tc.Resources)
```

Comment:
please see if ```ci.Resources.CPUReq = ""``` is really needed. if not, this can just be
```
if ci.Resources.CPUReq == "" {
        ci.Resources.CPUReq = DefaultMinVerifyCpuRequests
        defer func() {.    <<--- if we really need to set it back to ""
                ci.Resources.CPUReq = ""
        }
}
tc := makeTestContainer(ci)
...
```